File: src/main/java/uk/gov/justice/hmpps/casenotes/dto/UpdateCaseNoteType.java
package uk.gov.justice.hmpps.casenotes.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.Hidden;;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;

import javax.validation.constraints.NotBlank;

@Schema(description = "Update a Case Note Type")
@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@Data
public class UpdateCaseNoteType {

    @Schema(required = true, description = "Type Description", example = "General Note Type")
    @Length(max = 80)
    @NotBlank
    private String description;

    @Schema(description = "Active Type", example = "true")
    @Builder.Default
    private boolean active = true;

    @Schema(description = "Sensitive Case Note Type, default true", example = "true")
    @Builder.Default
    private boolean sensitive = true;

    @Schema(description = "Restricted Use, default true", example = "true")
    @Builder.Default
    private boolean restrictedUse = true;
}

File: src/main/java/uk/gov/justice/hmpps/casenotes/types/UpdateCaseNoteType.kt
package uk.gov.justice.hmpps.casenotes.types

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank
import org.hibernate.validator.constraints.Length

data class UpdateParentType(
  @Schema(required = true, description = "Type Description", example = "General Note Type")
  @field:Length(max = 80)
  @field:NotBlank
  val description: String,
)

data class UpdateSubType(
  @Schema(required = true, description = "Type Description", example = "General Note Type")
  @field:Length(max = 80)
  @field:NotBlank
  val description: String,

  @Schema(description = "Active Type, default true", example = "true")
  val active: Boolean = true,

  @Schema(description = "Sensitive Case Note Type, default true", example = "true")
  val sensitive: Boolean = true,

  @Schema(description = "Restricted Use, default true", example = "true")
  val restrictedUse: Boolean = true,
)